app/database/itemStore.go

```diff
@@ -106,21 +108,214 @@ func (s *ItemStore) GetRawNavigationData(rootID int64, userID, userLanguageID in
+	AccessRestricted 			*bool    `sql:"column:bAccessRestricted"`
+
+	*ItemAccessDetails
+}
+
+// GetRawItemData reads data needed by the getItem service from the DB and returns an array of RawItem's
+func (s *ItemStore) GetRawItemData(rootID, userID, userLanguageID int64, user AuthUser) (*[]RawItem, error) {
+	var result []RawItem
+
+	// This query can be simplified if we add a column for relation degrees into `items_ancestors`
+	if err := s.Raw(
+		"SELECT "+
+			"union_table.ID, "+
+			"union_table.sType, "+
+			"union_table.bDisplayDetailsInParent, "+
+			"union_table.sValidationType, "+
+			"COALESCE(union_table.idItemUnlocked, '')<>'' as hasUnlockedItems, "+
+			"union_table.iScoreMinUnlock, "+
+			"union_table.sTeamMode, "+
+			"union_table.bTeamsEditable, "+
+			"union_table.iTeamMaxMembers, "+
+			"union_table.bHasAttempts, "+
+			"union_table.sAccessOpenDate, "+
+			"union_table.sDuration, "+
+			"union_table.sEndContestDate, "+
+			"union_table.bNoScore, "+
+			"union_table.groupCodeEnter, "+
+
+			"COALESCE(ustrings.idLanguage, dstrings.idLanguage) AS idLanguage, " +
+			"IF(ustrings.idLanguage IS NULL, dstrings.sTitle, ustrings.sTitle) AS sTitle," +
+			"IF(ustrings.idLanguage IS NULL, dstrings.sImageUrl, ustrings.sImageUrl) AS sImageUrl, " +
+			"IF(ustrings.idLanguage IS NULL, dstrings.sSubtitle, ustrings.sSubtitle) AS sSubtitle, " +
+			"IF(ustrings.idLanguage IS NULL, dstrings.sDescription, ustrings.sDescription) AS sDescription, " +
+			"IF(ustrings.idLanguage IS NULL, dstrings.sEduComment, ustrings.sEduComment) AS sEduComment, " +
+
+			"users_items.idAttemptActive AS idAttemptActive," +
+			"users_items.iScore AS iScore, " +
+			"users_items.nbSubmissionsAttempts AS nbSubmissionsAttempts, " +
+			"users_items.bValidated AS bValidated, " +
+			"users_items.bFinished AS bFinished, " +
+			"users_items.bKeyObtained AS bKeyObtained, " +
+			"users_items.nbHintsCached AS nbHintsCached, " +
+			"users_items.sStartDate AS sStartDate, " +
+			"users_items.sValidationDate AS sValidationDate, " +
+			"users_items.sFinishDate AS sFinishDate, " +
+			"users_items.sContestStartDate AS sContestStartDate, " +
+			"IF(union_table.sType <> 'Chapter', users_items.sState, NULL) AS sState, " +
+			"IF(union_table.sType <> 'Chapter', users_items.sAnswer, NULL) AS sAnswer, " +
+
+			"union_table.iChildOrder AS iChildOrder, " +
+			"union_table.sCategory AS sCategory, " +
+			"union_table.bAlwaysVisible, " +
+			"union_table.bAccessRestricted, " +
+		// root node only
+			"union_table.bTitleBarVisible, " +
+			"union_table.bReadOnly, " +
+			"union_table.sFullScreen, " +
+			"union_table.bShowSource, " +
+			"union_table.iValidationMin, " +
+			"union_table.bShowUserInfos, " +
+			"union_table.sContestPhase, " +
+			"union_table.sUrl, " +
+			"union_table.bUsesAPI, " +
+			"union_table.bHintsAllowed, " +
+			"accessRights.fullAccess, accessRights.partialAccess, accessRights.grayedAccess, accessRights.accessSolutions "+
+			"FROM " +
```

Comment:
could you align the `FROM` with the initial `SELECT` so that it is easier to find the different part?
In the same way, for the next line, I would do something like:
```
FROM (
  SELECT 
    items.ID. ...
    ...
  FROM ...
)
```
etc with indent from the subqueries

Reply:
done